### PR TITLE
Migrate ISSM to GitHub repo

### LINKED ISSUE #167: ISSM: transition to GitHub repo
ISSM code was under subversion but is now hosted on GitHub

https://github.com/ISSMteam/ISSM

The spack package should be updated to use this as the code source

https://github.com/ACCESS-NRI/spack-packages/blob/main/packages/issm/package.py

### PR DESCRIPTION
**Pull request by aidanheerdegen**
Add support for new [ISSM GitHub repo](https://github.com/ISSMteam/ISSM).

Closes #167

### CODE REVIEW

**Comment by aidanheerdegen** [comment]
As noted in the issue, compiles successfully with this change

https://github.com/ACCESS-NRI/spack-packages/issues/167#issuecomment-2505240221

But does require `gcc/13`: `spack install issm %gcc@13

Should this also be in the package recipe until https://github.com/Reference-ScaLAPACK/scalapack/issues/97 is resolved?

File: packages/issm/package.py [hunk 1]
```diff
@@ -25,6 +26,11 @@ class Issm(AutotoolsPackage):
     depends_on("petsc+metis+mumps+scalapack")
     depends_on("m1qn3")
 
+    def url_for_version(self, version):
```
Comment (harshula) [inline packages/issm/package.py:29]: Does our usual syntax and github URL not work? i.e.
```
    def url_for_version(self, version):
        return "https://github.com/ISSMteam/ISSM/tarball/{0}".format(version)
```

File: packages/issm/package.py [hunk 2]
```diff
@@ -25,6 +26,11 @@ class Issm(AutotoolsPackage):
...
+        url = "https://github.com/ISSMteam/ISSM/archive/refs/tags/v{0}.tar.gz".format(version)
+        return url
```
Comment (harshula) [inline packages/issm/package.py:31]: Does our usual syntax and github URL not work? i.e.
```
        return "https://github.com/ISSMteam/ISSM/tarball/{0}".format(version)
```

Comment (aidanheerdegen) [reply packages/issm/package.py]: I don't think I tried that one. I did copy one from another package that was different again, and it definitely didn't work.

I have changed to the suggested format above and it works. Note that they're using `vX.X` as their tags, so I've made the change as above, but pre-pending a `v`.

**Review by harshula — APPROVED** [review]

**Pull request merged**